src/estimagic/estimate_ml.py

```diff
@@ -179,6 +184,9 @@ def estimate_ml(
     bounds = pre_process_bounds(bounds)
     jacobian_numdiff_options = pre_process_numdiff_options(jacobian_numdiff_options)
     hessian_numdiff_options = pre_process_numdiff_options(hessian_numdiff_options)
+    # TODO: Replace dict_constraints with constraints, once we deprecate dictionary
+    # constraints.
+    dict_constraints = pre_process_constraints(constraints)
```

Comment:
I would prefer to read `deprecations.pre_process_constraints` to make it even more visible that this is something that will need to be changed. I use `deprecations.func` almost everywhere instead of importing those functions one by one.